### PR TITLE
Fix up element/elements/section/sections to all contain correct tests

### PR DESCRIPTION
**Pull request by luke-hill**
As well as that I've introduced some additional xpath tests where appropriate to duplicate the test coverage where we perhaps were a bit light.

### CODE REVIEW

**Comment by luke-hill** [comment]
Ping @TheMetalCode

**Comment by luke-hill** [comment]
As I think you're back from hols I may as well ping you as well @tgaff

**Comment by tgaff** [comment]
Back in just a couple days actually.  I'll try to take a look soon.

File: spec/element_spec.rb [hunk 1]
```diff
@@ -3,105 +3,68 @@
...
 
-  it 'element method without css should generate existence check method' do
-    class PageWithElement < SitePrism::Page
-      element :thing, 'input#nonexistent'
-    end
-    page = PageWithElement.new
-    expect(page).to respond_to :has_no_thing?
-  end
+    describe '#expected_elements' do
+      subject { page.all_there? }
```
[the remark this answers is not in the document]
Comment (luke-hill) [reply spec/element_spec.rb]: Yes it is whoops :)

File: spec/element_spec.rb [hunk 2]
```diff
@@ -3,105 +3,68 @@
...
+      end
     end
-    page = PageWithElement.new
-    expect(page).to respond_to :wait_for_some_slow_element
   end
 
-  it 'should know if all mapped elements are on the page' do
-    class PageWithAFewElements < SitePrism::Page
+  context 'with a css element' do
+    class Page < SitePrism::Page
```
[the remark this answers is not in the document]
Comment (luke-hill) [reply spec/element_spec.rb]: Yes we could probably. I'll refactor this. Good spot. (I'll do this for both)

**Comment by luke-hill** [comment]
I've fixed up this branch and fixed up most of your comments @tgaff. I've also added extra tests where appropriate from the work from @ricmatsui and kept as much intact as possible.

In adding the extra tests. RSpec started to mix the two shared contexts together (`expected_elements` became duplicated). So what I did is called the classes distinct names, and then referenced the verbose classname in the classmethod description by doing `klass.expected_elements`

**Comment by luke-hill** [comment]
I'd like to merge this PR in next (As this is the one that is most likely to conflict with other stuff), as it's the last "big" refactor of spec code (There is another PR in the works which refactors a load of cucumber code as well). And then cut 2.13 as the last big "refactor heavy" release

cc/ @tgaff

**Pull request merged**